### DISCUSSION TITLE
handlebar question

### DISCUSSION THREAD

**Discussion by BillThyCat** [Budibase/budibase#6359] [Help]
Our pay periods are Wed-Tue each week.  I'm working on a timesheet app and I want to set a constraint on the date they choose to put in time for to not be earlier than the previous Wednesday since we don't want them entering time for previous weeks, just the current pay week.

How would I calculate the previous Wednesday's date using handlebars?  As a workaround I could just pick today's date minus 7, but that could allow someone to sneak in at the end of the previous weeks pay period.

Thanks for any help you can provide.

**Comment by melohagan** [comment]
Might be easier to use a JavaScript binding for this one.
Something like:
```
let daysFromLastWednesday = new Date().getDay() - 3;
if (daysFromLastWednesday < 0) {
  daysFromLastWednesday += 7;
}
let lastWednesday = new Date();
lastWednesday.setDate(lastWednesday.getDate() - daysFromLastWednesday);
return lastWednesday;
```

I haven't tested this but I think that's roughly right.

1. *Get the difference in days between today and Wednesday (3)*
`let daysFromLastWednesday = new Date().getDay() - 3;`

3. *If the number is negative, it means the previous wednesday was in the previous week, so add 7 days*
```
if (daysFromLastWednesday < 0) {
  daysFromLastWednesday += 7;
}
```

4. *Return todays date minus the days to the calculated previous Wednesday*
```
let lastWednesday = new Date();
lastWednesday.setDate(lastWednesday.getDate() - daysFromLastWednesday);
return lastWednesday;
```

----
I imagine this could be done with Handlebars, but it would probably be tricky to read, although that may simply be personal preference as well.

**Reply by BillThyCat** [reply]
Ok, the formula works to calculate the previous Wednesday.  Thank you!

Now a side question relating to this question.

Lets say instead of it grabbing todays date with the function Date(), how do I pass a field from the form to the JavaScript side?  Do I put it in the handlebar section (the field data), then run the js function against it?

i.e. I have a date picker field to select the day of the week you want to enter time for, I will use your function for the validation part to make sure the date is within the week.  Now I would like to calculate the day of week based on the date they pick.  This will be a different field in the same form.  So how do I get javascript to be able to look into a different field?  Or can that be done?

Thanks again!

**Comment by melohagan** [comment]
@BillThyCat So I think what you want is to configure validation:

**Click configure validation**
![Screenshot 2022-06-17 at 14 57 53](https://user-images.githubusercontent.com/101575380/174312910-1ebd28a5-5f6e-467c-a8b7-f94bee44a114.png)

![Screenshot 2022-06-17 at 14 58 24](https://user-images.githubusercontent.com/101575380/174312994-7a6fe670-5541-40dd-9287-c4ee18a26706.png)
The JavaScript binding is the code I posted above.

**Make sure the Save button has a Validate Form action**
![Screenshot 2022-06-17 at 14 59 10](https://user-images.githubusercontent.com/101575380/174313134-0aa8dc93-b8c5-4337-97b1-c8d5666830d6.png)

**Result**
![validatewed](https://user-images.githubusercontent.com/101575380/174313336-c15ec130-2e5f-47c3-84d9-36f94c004dac.gif)
